pkg/deb/DEBIAN/postinst

```diff
@@ -1,5 +1,5 @@
-#!/bin/sh
+#!/bin/sh -e
 USER=peerbanhelper
-adduser --system $USER
-mkdir -p /etc/peerbanhelper
-chown $USER /etc/peerbanhelper
\ No newline at end of file
+getent passwd peerbanhelper > /dev/null || adduser --system --group --no-create-home --disabled-password --home /etc/peerbanhelper $USER
+chown -R $USER: /etc/peerbanhelper
+systemctl enable --now peerbanhelper.service 2> /dev/null
```

Comment:
Auto start? I don't think this should be written in the post-install script, this should be executed by user

Reply:
Why? PBH starts with a wizard anyway (doesn't do anything without configuration) and the user might not know that they need to `systemctl start` the service. The only thing that would suck is if 9898 is already in use making the installation not "finish successfully" but since it's the last command anyway it might not be that big of a deal.

Reply:
I think @Gaojianli would like us to leave it unchanged and let the end user decide and choose whether PeerBanHelper's service should be started automatically or not.
This is not a problem for professional Linux operations, but considering that PeerBanHelper users may simply install the package on their HomeLab or NAS, this action is worthwhile in my opinion.

Reply:
the systemd is set with `Restart=on-failure`, which means if 9898 is not work, the pbh will keep restarting and drain the resource of the system. Anyway, please remove this. For user who don't know `systemctl start`, we can add a message to notice them, just like the freebsd version: 
https://github.com/PBH-BTN/PeerBanHelper/blob/21ea4eac44763979b3f30eb9bbe8afb75aeae0b9/pkg/pkg/%2BDISPLAY#L2

Reply:
> which means if 9898 is not work, the pbh will keep restarting and drain the resource of the system.

But only up to 5 times then it will enter the failed state.

> Anyway, please remove this.

What exactly? The starting of the service, the enabling (start at boot but not right now) or both?

Reply:
@Anuskuss I checked this script and I noticed an issue, if the user disables the service autostart after installing the package, this script seems to re-enable the service autostart when upgrading the package.

Is it possible to enable autostart only on the first installation?   

I would like the install script to honor the user's choice. Because it is the user's right to cancel autostart. So we shouldn't override it while upgrading.

Reply:
> > which means if 9898 is not work, the pbh will keep restarting and drain the resource of the system.
> 
> But only up to 5 times then it will enter the failed state.
> 
> > Anyway, please remove this.
> 
> What exactly? The starting of the service, the enabling (start at boot but not right now) or both?

do nothing but 
```
echo "To start peerbanhelper, please run 'systemctl start peerbanhelper.service'"
echo "To auto-start, please run 'systemctl enable peerbanhelper.service'"
```

Reply:
> Is it possible to enable autostart only on the first installation?

Yeah sure you can filter by event so on `install` you enable and on `upgrade` you do nothing.

> do nothing

Sure I can do that. Hopefully the user reads this though but I guess Linux users really don't need much hand-holding. Will implement this once I get back from work.

Reply:
> > Is it possible to enable autostart only on the first installation?
> 
> Yeah sure you can filter by event so on `install` you enable and on `upgrade` you do nothing.
> 
> > do nothing
> 
> Sure I can do that. Hopefully the user reads this though but I guess Linux users really don't need much hand-holding. Will implement this once I get back from work.

We are discussing the changes regarding autostart and will get in touch with you when we have new results.

Reply:
> > Is it possible to enable autostart only on the first installation?
> 
> Yeah sure you can filter by event so on `install` you enable and on `upgrade` you do nothing.
> 
> > do nothing
> 
> Sure I can do that. Hopefully the user reads this though but I guess Linux users really don't need much hand-holding. Will implement this once I get back from work.

can you ref this script?
```bash
SERVICE="peerbanhelper.service"
INSTALLED_FLAG="/etc/peerbanhelper/.installed"
if [ -e $INSTALLED_FLAG ]; then
    # update
    if systemctl is-active --quiet "$SERVICE" > /dev/null 2>&1; then
        echo "$SERVICE is already running. Restarting the service."
        systemctl restart "$SERVICE"
    fi
else
    # first install
    systemctl enable --now $SERVICE
    touch $INSTALLED_FLAG
fi
```

Reply:
Autostart is gone now.